Base/Python/slicer/util.py

```diff
@@ -1977,6 +1977,152 @@ def updateTransformMatrixFromArray(transformNode, narray, toWorld=False):
+    itkImage.SetOrigin(origin)
+    itkImage.SetSpacing(spacing)
+    directionMatrix = np.asarray(directionTuple).reshape((3, 3))
+    itkImage.SetDirection(itk.matrix_from_array(directionMatrix))
+
+    return itkImage
+
+
+def _getRASAffineArrayFromITKImage(itkImage):
+    """
+    Adapted from TorchIO. See:
+    https://github.com/fepegar/torchio/blob/4c1b3d83a7962699a15afe76ae6f39db1aae7a99/src/torchio/data/io.py#L356-L381
+    """
+
+    import numpy as np
+
+    spacing = np.array(itkImage.GetSpacing())
+    direction_lps = np.array(itkImage.GetDirection())
+    origin_lps = np.array(itkImage.GetOrigin())
+    direction_length = len(direction_lps)
+
+    if itkImage.ndim == 3:
+        rotation_lps = direction_lps.reshape(3, 3)
+
+    elif itkImage.ndim == 2:  # ignore last dimension if 2D (1, W, H, 1)
+        rotation_lps_2d = direction_lps.reshape(2, 2)
+        rotation_lps = np.eye(3)
+        rotation_lps[:2, :2] = rotation_lps_2d
+        spacing = np.append(spacing, 1)
+        origin_lps = np.append(origin_lps, 0)
+
+    elif itkImage.ndim == 4:  # probably a bad NIfTI. Let's try to fix it
+        rotation_lps = direction_lps.reshape(4, 4)[:3, :3]
+        spacing = spacing[:-1]
+        origin_lps = origin_lps[:-1]
+
+    FLIPXY_33 = np.diag([-1, -1, 1])  # Matrix used to switch between LPS and RAS
+    rotation_ras = np.dot(FLIPXY_33, rotation_lps)
+    rotation_ras_zoom = rotation_ras * spacing
+    translation_ras = np.dot(FLIPXY_33, origin_lps)
+    affine = np.eye(4)
+    affine[:3, :3] = rotation_ras_zoom
+    affine[:3, 3] = translation_ras
+    return affine
+
+
+def addVolumeFromITKImage(itkImage, name=None):
+    """Create a new volume node from content of an ITK image and add it to the scene.
+
+    :param itkImage: ITK image containing volume voxels.
+    :param name: volume node name
+    :return: created new volume node
+    """
+    if name is None:
+        name = ""
+    volumeNode = slicer.mrmlScene.AddNewNodeByClass("vtkMRMLScalarVolumeNode", name)
+    updateVolumeFromITKImage(volumeNode, itkImage)
+    return volumeNode
+
+
+def updateVolumeFromITKImage(volumeNode, itkImage):
+    """Set ITK image to volume node and display it."""
+    import itk
+    import numpy as np
+    rasAffine = _getRASAffineArrayFromITKImage(itkImage)
+    ijkToRAS = vtkMatrixFromArray(rasAffine)
+    vtkImage = itk.vtk_image_from_image(itkImage)
+
+    # set identity metadata on the vtkImageData for the volume node
+    # otherwise display properties are bugged, see:
+    # https://github.com/Slicer/Slicer/issues/6911
+    vtkImage.SetSpacing([1.0] * itkImage.ndim)
+    vtkImage.SetOrigin([0.0] * itkImage.ndim)
+    vtkImage.SetDirectionMatrix(np.eye(itkImage.ndim).flatten())
+    volumeNode.SetAndObserveImageData(vtkImage)
+    volumeNode.SetIJKToRASMatrix(ijkToRAS)
+
+    setSliceViewerLayers(
+        background=volumeNode,
+        fit=True,
+        rotateToVolumePlane=True,
+        )
```

Comment:
For consistency with existing API, I think we should remove this and instead ensure that the volume node invoke the relevant modified events.